Add flags to print CLI usage help message and version

jessicasyu opened the pull request:
Updated command line to support `--version` (which prints the version) and `-h/--help` (which prints the help message listing usage, arguments, and options (following [docopt](http://docopt.org) conventions).

## Summary of changes

- Add `--version` command to print version
- Add `-h` and `--help` commands to print help message
- Update command names and help messages for clarity 
- Update `Command` class (in `InputParser`) to support docopt-style usage and descriptions

## Steps to verify

1. Build jar
2. Run `java -jar arcade.jar --version` (displays version)
3. Run `java -jar arcade.jar -h` (displays help message)
4. Run `java -jar arcade.jar --help` (displays help message)

Current setup is that including `-h` or `--help` anywhere in the command will print the help message. Alternative would be to support only the options shown in step 3 and step 4 (putting the help command anywhere else would throw an error). Not sure which is more intuitive.

Jannetty commented:
This all looks great to me, super helpful features! Would it be worth prompting users to use the -h or --help options when someone uses a flag that doesn't exist? So for example if I run with --h accidentally (or otherwise run with an unsupported flag), I get

```
2024-12-16 15:24:33 WARNING arcade.core.ARCADE                  : ARCADE [ --h ] does not exist
Exception in thread "main" java.security.InvalidParameterException
	at arcade.core.ARCADE.main(ARCADE.java:131)
	at arcade.core.gui.GUI.main(GUI.java:80)
```

Would it be worth adding a message saying "use --help for usage instructions" to the warning that is raised when a flag that doesn't exist is used?

jessicasyu commented:
@Jannetty Good suggestion! I added some help messages for the parser. I also updated the specific message you pointed out to be more clear what the valid options are. There is probably a cleaner way to do this, instead of just appending text, but logging is pretty clunky right now and we need to revisit
